dll/win32/shell32/utils.cpp

```diff
@@ -99,3 +99,134 @@ SHFindComputer(LPCITEMIDLIST pidlRoot, LPCITEMIDLIST pidlSavedSearch)
+#if (WINVER >= 0x0600)
+    BOOL bMinus = (llValue < 0);
+
+    if (bMinus)
+        llValue = -llValue;
+#endif
+
+    do
+    {
+        szBuff[ich++] = (WCHAR)(L'0' + (llValue % 10));
+        llValue /= 10;
+    } while (llValue != 0);
+
+#if (WINVER >= 0x0600)
+    if (bMinus)
+        szBuff[ich++] = '-';
+#endif
+
+    do
+    {
+        --ich;
+        *pszValue++ = szBuff[ich];
+    } while (ich > 0);
+
+    *pszValue = UNICODE_NULL;
+}
+
+#define MAX_NUMSEP 6
+
+static VOID
+Int64GetNumFormat(NUMBERFMTW *pDest, const NUMBERFMTW *pSrc, DWORD dwNumberFlags,
+                  LPWSTR pszDecimal, LPWSTR pszThousand)
+{
+    WCHAR szBuff[20];
+
+    if (pSrc)
+        *pDest = *pSrc;
+    else
+        dwNumberFlags = 0;
+
+    if (!(dwNumberFlags & NUMFLAG_NUMDIGITS))
+    {
+        GetLocaleInfoW(LOCALE_USER_DEFAULT, LOCALE_IDIGITS, szBuff, _countof(szBuff));
+        pDest->NumDigits = StrToIntW(szBuff);
+    }
+
+    if (!(dwNumberFlags & NUMFLAG_LEADINGZERO))
+    {
+        GetLocaleInfoW(LOCALE_USER_DEFAULT, LOCALE_ILZERO, szBuff, _countof(szBuff));
+        pDest->LeadingZero = StrToIntW(szBuff);
+    }
+
+    if (!(dwNumberFlags & NUMFLAG_GROUPING))
+    {
+        GetLocaleInfoW(LOCALE_USER_DEFAULT, LOCALE_SGROUPING, szBuff, _countof(szBuff));
+        pDest->Grouping = StrToIntW(szBuff);
+    }
+
+    if (!(dwNumberFlags & NUMFLAG_DECIMALSEP))
+    {
+        GetLocaleInfoW(LOCALE_USER_DEFAULT, LOCALE_SDECIMAL, pszDecimal, MAX_NUMSEP);
+        pDest->lpDecimalSep = pszDecimal;
+    }
+
+    if (!(dwNumberFlags & NUMFLAG_THOUSANDSEP))
+    {
+        GetLocaleInfoW(LOCALE_USER_DEFAULT, LOCALE_STHOUSAND, pszThousand, MAX_NUMSEP);
+        pDest->lpThousandSep = pszThousand;
+    }
+
+    if (!(dwNumberFlags & NUMFLAG_NEGATIVEORDER))
+    {
+        GetLocaleInfoW(LOCALE_USER_DEFAULT, LOCALE_INEGNUMBER, szBuff, _countof(szBuff));
+        pDest->NegativeOrder = StrToIntW(szBuff);
+    }
+}
+
+/*************************************************************************
+ *  Int64ToString [SHELL32.209]
+ */
+EXTERN_C
+INT WINAPI
+Int64ToString(
+    LONGLONG llValue,
+    LPWSTR pszOut,
+    UINT cchOut,
+    BOOL bUseFormat,
+    NUMBERFMTW *pNumberFormat,
+    DWORD dwNumberFlags)
+{
+    INT ret;
+    NUMBERFMTW NumFormat;
+    WCHAR szValue[80], szDecimalSep[MAX_NUMSEP], szThousandSep[MAX_NUMSEP];
+
+    Int64ToStr(llValue, szValue);
+
+    if (bUseFormat)
+    {
+        Int64GetNumFormat(&NumFormat, pNumberFormat, dwNumberFlags, szDecimalSep, szThousandSep);
+        ret = GetNumberFormatW(LOCALE_USER_DEFAULT, 0, szValue, &NumFormat, pszOut, cchOut);
+        if (ret)
+            --ret;
+        return ret;
+    }
+
+    if (FAILED(StringCchCopyW(pszOut, cchOut, szValue)))
+        return 0;
+
+    return lstrlenW(pszOut);
+}
+
+/*************************************************************************
+ *  LargeIntegerToString [SHELL32.210]
+ */
+EXTERN_C
+INT WINAPI
+LargeIntegerToString(LARGE_INTEGER *pLargeInt,
```

Comment:
```suggestion
LargeIntegerToString(
    LARGE_INTEGER *pLargeInt,
```
Also, you can simplify this with `PLARGE_INTEGER`.

In addition you can restyle similarly the parameters list of the other new functions you've added in this file. What about also adding SAL annotations?

Reply:
Added SAL.
Fixed prototypes (I added some `const`).